init.lua

```diff
@@ -137,23 +141,64 @@ function can_build_rail(start_pos, end_pos)
 	return false
 end
 
+local function can_dig(pos, player)
+	return (minetest.registered_nodes[advtrains.ndb.get_node(pos).name].can_dig or function() return true end)(pos, player)
+end
+
+--[[
+Checks if it is "appropriate" to change the node at <pos> to <newnode>
+Returns: new node (may be adjusted for e.g. turnouts or crossings), error message for the player
+--]]
+local function can_overwrite_track_at(player, pos, newnode)
+	local oldnode = advtrains.ndb.get_node(pos)
+	local oldname, oldparam2, newname, newparam2 = oldnode.name, oldnode.param2, newnode.name, newnode.param2
+	if not can_dig(pos, player) then
+		return false, S("Cannot modify the node at @1", minetest.pos_to_string(pos))
+	end
+	if oldname == newname and oldparam2 == newparam2 then
+		return false
+	end
+	if not advtrains.is_track_and_drives_on(oldname) then
+		return newnode
+	end
+	if not advtrains.is_track_and_drives_on(newname) then
+		return false, S("@1 is not a track", quote_string(newname))
+	end
+	local oldconns, _, oldtype = advtrains.get_track_connections(oldname, oldparam2)
+	local newconns, _, newtype = advtrains.get_track_connections(newname, newparam2)
+	if oldtype ~= newtype then
+		return false, S("The track at @1 is of a different type than @2", minetest.pos_to_string(pos), quote_string(newname))
+	elseif #oldconns > #newconns then
+		return false, S("@1 would not fit the track at @2", quote_string(newname), minetest.pos_to_string(pos))
+	end
+	if oldconns[3] and newconns[3] and type(oldconns[5]) ~= type(newconns[5]) then
+		return false, S("@1 would not fit the track at @2", quote_string(newname), minetest.pos_to_string(pos))
+	end
+	return newnode
+end
+
+local function try_build(player, pos, newnode)
+	local place_current, errmsg = can_overwrite_track_at(player, pos, newnode)
+	if place_current then
+		advtrains.ndb.swap_node(pos, place_current)
+	end
+	if errmsg then
+		minetest.chat_send_player(player:get_player_name(), errmsg)
+	end
+	return place_current
+end
+
 -- build a rail with a fixed direction
 function build_rail(player, start_pos, end_pos, build_last_node)
 	local delta_pos = vector.subtract(end_pos, start_pos)
 	local rail_node_params_seq = advtrain_helpers.direction_step_to_rail_params_sequence(delta_pos)
 	local direction_delta = delta_to_dir(delta_pos)
 	local current_pos = table.copy(start_pos)
-	current_pos.y = current_pos.y - 0.2 -- decrement a little bit, because y values will be rounded for slope placement
```

Comment:
I have (temporarily) removed this because Advtrains' node database API seems to have some problems with coordinates whose elements are not integers. I did read the commend and will put a (hopefully) better mechanism in place for slopes.

Reply:
I have fixed the regression of breaking slope building (see 9d072a6 and 27b22af). The remaining issues can be observed upstream and, as I mentioned in the first post, cannot be fixed on my own.